index.js

```diff
@@ -64,10 +65,16 @@ const slugify = (string, options) => {
 		patternSlug = /[^a-z\d]+/g;
 	}
 
+	const shouldPrependUnderscore = options.preserveLeadingUnderscore && string.startsWith('_');
```

Comment:
This check should be done before any string transformations are done.

Reply:
Indeed! Done.